src/engine/x86-64/X86_64Interpreter.v3

```diff
@@ -2517,6 +2542,18 @@ class X86_64InterpreterGen(ic: X86_64InterpreterCode, w: DataWriter) {
 			asm.movdqu_m_s(vsph[-1].value, r_xmm0);
 			endHandler();
 		}
+		for (t in [
+			(Opcode.I8X16_EXTRACTLANE_U, BpTypeCode.I32.code, byte.!(1), asm.movb_r_m, asm.movb_m_r, false),
+			(Opcode.I16X8_EXTRACTLANE_U, BpTypeCode.I32.code, byte.!(2), asm.movw_r_m, asm.movw_m_r, false),
+			(Opcode.I8X16_EXTRACTLANE_S, BpTypeCode.I32.code, byte.!(1), asm.movb_r_m, asm.movd_m_r, true),
+			(Opcode.I16X8_EXTRACTLANE_S, BpTypeCode.I32.code, byte.!(2), asm.movw_r_m, asm.movd_m_r, true),
+			(Opcode.I32X4_EXTRACTLANE, BpTypeCode.I32.code, byte.!(4), asm.movd_r_m, asm.movd_m_r, false),
+			(Opcode.I64X2_EXTRACTLANE, BpTypeCode.I64.code, byte.!(8), asm.movq_r_m, asm.movq_m_r, false),
+			(Opcode.F32X4_EXTRACTLANE, BpTypeCode.F32.code, byte.!(4), asm.movd_r_m, asm.movd_m_r, false),
+			(Opcode.F64X2_EXTRACTLANE, BpTypeCode.F64.code, byte.!(8), asm.movq_r_m, asm.movq_m_r, false)
+		]) {
+			genExtractLane(t);
```

Comment:
I think it would be clearer to just have 8 calls to genExtractLane, rather than a loop over the tuples.